compiler/rustc_metadata/src/locator.rs

```diff
@@ -654,7 +654,22 @@ impl<'a> CrateLocator<'a> {
                     continue;
                 }
             }
-            *slot = Some((hash, metadata, lib.clone()));
+
+            // Question: maybe we shouldn't error eagerly here, because it is possible that
+            // e.g. a rlib only contains a stub, but a (later-resolved) dylib contains the full
```

Comment:
`-Zsplit-metadata` applies to all outputs. And both rlib and dylib must be compiled by the same rustc invocation to be guaranteed that the crate hash matches (and thus they are considered interchangeable in dependency resolution).

Reply:
Ok, that makes sense. Thanks!